test/unit/image-transform.spec.ts

```diff
@@ -0,0 +1,253 @@
+      disable: 'upscale',
+      width: '200',
+      height: '200',
+    });
+  });
+
+  it('should return valid object when crop method is called with valid params', () => {
+    expect(getBuild(new ImageTransform().crop({ width: 100, height: 200 }))).toEqual({ crop: ['100', '200'] });
+    expect(getBuild(new ImageTransform().crop({ width: 2, height: 3, cropBy: CropBy.ASPECTRATIO }))).toEqual({
+      crop: '2:3',
+    });
+    expect(
+      getBuild(new ImageTransform().crop({ width: 200, height: 300, cropBy: CropBy.REGION, xval: 100, yval: 150 }))
+    ).toEqual({
+      crop: ['200', '300', 'x100', 'y150'],
+    });
+    expect(
+      getBuild(new ImageTransform().crop({ width: 200, height: 300, cropBy: CropBy.OFFSET, xval: 100, yval: 150 }))
+    ).toEqual({
+      crop: ['200', '300', 'offset-x100', 'offset-y150'],
+    });
+    expect(
+      getBuild(
+        new ImageTransform().crop({
+          width: 200,
+          height: 300,
+          cropBy: CropBy.REGION,
+          xval: 100,
+          yval: 150,
+          safe: true,
+        })
+      )
+    ).toEqual({
+      crop: ['200', '300', 'x100', 'y150', 'safe'],
+    });
+    expect(
+      getBuild(
+        new ImageTransform().crop({
+          width: 200,
+          height: 300,
+          cropBy: CropBy.REGION,
+          xval: 100,
+          yval: 150,
+          smart: true,
+        })
+      )
+    ).toEqual({
+      crop: ['200', '300', 'x100', 'y150', 'smart'],
+    });
+    expect(
+      getBuild(
+        new ImageTransform().crop({
+          width: 200,
+          height: 300,
+          cropBy: CropBy.REGION,
+          xval: 100,
+          yval: 150,
+          smart: true,
+          safe: true,
+        })
+      )
+    ).toEqual({
+      crop: ['200', '300', 'x100', 'y150', 'safe', 'smart'],
+    });
+  });
+
+  it('should return valid object when fit method is called with valid params', () => {
+    expect(getBuild(new ImageTransform().fit(FitBy.BOUNDS))).toEqual({ fit: 'bounds' });
+    expect(getBuild(new ImageTransform().fit(FitBy.CROP))).toEqual({ fit: 'crop' });
+  });
+
+  it('should return valid object when trim method is called with valid params', () => {
+    expect(getBuild(new ImageTransform().trim([25, 50, 75, 90]))).toEqual({ trim: '25,50,75,90' });
+    expect(getBuild(new ImageTransform().trim([25, 50, 25]))).toEqual({ trim: '25,50,25' });
+    expect(getBuild(new ImageTransform().trim(50))).toEqual({ trim: '50' });
+  });
+
+  it('should return valid object when orient method is called with valid params', () => {
+    const imgTObj = getBuild(new ImageTransform().orient(Orientation.FLIP_HORIZONTAL));
+    expect(imgTObj).toEqual({ orient: '2' });
+  });
+
+  it('should return valid object when overlay method is called with valid params', () => {
+    const overlayImgURL = '/v3/assets/blteae40eb499811073/bltb21dacdd20d0e24c/59e0c401462a293417405f34/circle.png';
```

Comment:
remove the credentials from the file @nadeem-cs

Reply:
Resolved in [this](https://github.com/contentstack/contentstack-javascript/pull/110) PR.